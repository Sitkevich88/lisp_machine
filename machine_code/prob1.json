[
    {
        "opcode": "loadc",
        "arg": 0,
        "term": 0
    },
    {
        "opcode": "store",
        "arg": 3,
        "term": 1
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 2
    },
    {
        "opcode": "loadc",
        "arg": 1,
        "term": 3
    },
    {
        "opcode": "store",
        "arg": 2,
        "term": 4
    },
    {
        "opcode": "store",
        "arg": 4,
        "term": 5
    },
    {
        "opcode": "loadc",
        "arg": 0,
        "term": 6
    },
    {
        "opcode": "store",
        "arg": 1,
        "term": 7
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 8
    },
    {
        "opcode": "loadc",
        "arg": 1,
        "term": 9
    },
    {
        "opcode": "store",
        "arg": 0,
        "term": 10
    },
    {
        "opcode": "store",
        "arg": 4,
        "term": 11
    },
    {
        "opcode": "load",
        "arg": 0,
        "term": 12
    },
    {
        "opcode": "store",
        "arg": 4,
        "term": 13
    },
    {
        "opcode": "load",
        "arg": 1,
        "term": 14
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 15
    },
    {
        "opcode": "load",
        "arg": 5,
        "term": 16
    },
    {
        "opcode": "store",
        "arg": 6,
        "term": 17
    },
    {
        "opcode": "addc",
        "arg": -3,
        "term": 18
    },
    {
        "opcode": "jl",
        "arg": 21,
        "term": 19
    },
    {
        "opcode": "jmp",
        "arg": 17,
        "term": 20
    },
    {
        "opcode": "load",
        "arg": 6,
        "term": 21
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 22
    },
    {
        "opcode": "load",
        "arg": 5,
        "term": 23
    },
    {
        "opcode": "addc",
        "arg": 0,
        "term": 24
    },
    {
        "opcode": "je",
        "arg": 28,
        "term": 25
    },
    {
        "opcode": "loadc",
        "arg": 0,
        "term": 26
    },
    {
        "opcode": "jmp",
        "arg": 29,
        "term": 27
    },
    {
        "opcode": "loadc",
        "arg": 1,
        "term": 28
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 29
    },
    {
        "opcode": "loadc",
        "arg": 2,
        "term": 30
    },
    {
        "opcode": "store",
        "arg": 4,
        "term": 31
    },
    {
        "opcode": "load",
        "arg": 4,
        "term": 32
    },
    {
        "opcode": "addc",
        "arg": -2,
        "term": 33
    },
    {
        "opcode": "jne",
        "arg": 37,
        "term": 34
    },
    {
        "opcode": "load",
        "arg": 5,
        "term": 35
    },
    {
        "opcode": "je",
        "arg": 53,
        "term": 36
    },
    {
        "opcode": "load",
        "arg": 2,
        "term": 37
    },
    {
        "opcode": "store",
        "arg": 4,
        "term": 38
    },
    {
        "opcode": "load",
        "arg": 3,
        "term": 39
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 40
    },
    {
        "opcode": "load",
        "arg": 0,
        "term": 41
    },
    {
        "opcode": "store",
        "arg": 6,
        "term": 42
    },
    {
        "opcode": "load",
        "arg": 1,
        "term": 43
    },
    {
        "opcode": "store",
        "arg": 7,
        "term": 44
    },
    {
        "opcode": "load",
        "arg": 5,
        "term": 45
    },
    {
        "opcode": "add",
        "arg": 7,
        "term": 46
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 47
    },
    {
        "opcode": "load",
        "arg": 4,
        "term": 48
    },
    {
        "opcode": "store",
        "arg": 2,
        "term": 49
    },
    {
        "opcode": "load",
        "arg": 5,
        "term": 50
    },
    {
        "opcode": "store",
        "arg": 3,
        "term": 51
    },
    {
        "opcode": "jmp",
        "arg": 94,
        "term": 52
    },
    {
        "opcode": "load",
        "arg": 0,
        "term": 53
    },
    {
        "opcode": "store",
        "arg": 4,
        "term": 54
    },
    {
        "opcode": "load",
        "arg": 1,
        "term": 55
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 56
    },
    {
        "opcode": "load",
        "arg": 5,
        "term": 57
    },
    {
        "opcode": "store",
        "arg": 6,
        "term": 58
    },
    {
        "opcode": "addc",
        "arg": -5,
        "term": 59
    },
    {
        "opcode": "jl",
        "arg": 62,
        "term": 60
    },
    {
        "opcode": "jmp",
        "arg": 58,
        "term": 61
    },
    {
        "opcode": "load",
        "arg": 6,
        "term": 62
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 63
    },
    {
        "opcode": "load",
        "arg": 5,
        "term": 64
    },
    {
        "opcode": "addc",
        "arg": 0,
        "term": 65
    },
    {
        "opcode": "je",
        "arg": 69,
        "term": 66
    },
    {
        "opcode": "loadc",
        "arg": 0,
        "term": 67
    },
    {
        "opcode": "jmp",
        "arg": 70,
        "term": 68
    },
    {
        "opcode": "loadc",
        "arg": 1,
        "term": 69
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 70
    },
    {
        "opcode": "loadc",
        "arg": 2,
        "term": 71
    },
    {
        "opcode": "store",
        "arg": 4,
        "term": 72
    },
    {
        "opcode": "load",
        "arg": 4,
        "term": 73
    },
    {
        "opcode": "addc",
        "arg": -2,
        "term": 74
    },
    {
        "opcode": "jne",
        "arg": 78,
        "term": 75
    },
    {
        "opcode": "load",
        "arg": 5,
        "term": 76
    },
    {
        "opcode": "je",
        "arg": 94,
        "term": 77
    },
    {
        "opcode": "load",
        "arg": 2,
        "term": 78
    },
    {
        "opcode": "store",
        "arg": 4,
        "term": 79
    },
    {
        "opcode": "load",
        "arg": 3,
        "term": 80
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 81
    },
    {
        "opcode": "load",
        "arg": 0,
        "term": 82
    },
    {
        "opcode": "store",
        "arg": 6,
        "term": 83
    },
    {
        "opcode": "load",
        "arg": 1,
        "term": 84
    },
    {
        "opcode": "store",
        "arg": 7,
        "term": 85
    },
    {
        "opcode": "load",
        "arg": 5,
        "term": 86
    },
    {
        "opcode": "add",
        "arg": 7,
        "term": 87
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 88
    },
    {
        "opcode": "load",
        "arg": 4,
        "term": 89
    },
    {
        "opcode": "store",
        "arg": 2,
        "term": 90
    },
    {
        "opcode": "load",
        "arg": 5,
        "term": 91
    },
    {
        "opcode": "store",
        "arg": 3,
        "term": 92
    },
    {
        "opcode": "jmp",
        "arg": 94,
        "term": 93
    },
    {
        "opcode": "load",
        "arg": 0,
        "term": 94
    },
    {
        "opcode": "store",
        "arg": 4,
        "term": 95
    },
    {
        "opcode": "load",
        "arg": 1,
        "term": 96
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 97
    },
    {
        "opcode": "load",
        "arg": 5,
        "term": 98
    },
    {
        "opcode": "addc",
        "arg": 1,
        "term": 99
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 100
    },
    {
        "opcode": "load",
        "arg": 4,
        "term": 101
    },
    {
        "opcode": "store",
        "arg": 0,
        "term": 102
    },
    {
        "opcode": "load",
        "arg": 5,
        "term": 103
    },
    {
        "opcode": "store",
        "arg": 1,
        "term": 104
    },
    {
        "opcode": "load",
        "arg": 0,
        "term": 105
    },
    {
        "opcode": "store",
        "arg": 4,
        "term": 106
    },
    {
        "opcode": "load",
        "arg": 1,
        "term": 107
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 108
    },
    {
        "opcode": "load",
        "arg": 5,
        "term": 109
    },
    {
        "opcode": "addc",
        "arg": -999,
        "term": 110
    },
    {
        "opcode": "jg",
        "arg": 114,
        "term": 111
    },
    {
        "opcode": "loadc",
        "arg": 0,
        "term": 112
    },
    {
        "opcode": "jmp",
        "arg": 115,
        "term": 113
    },
    {
        "opcode": "loadc",
        "arg": 1,
        "term": 114
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 115
    },
    {
        "opcode": "loadc",
        "arg": 2,
        "term": 116
    },
    {
        "opcode": "store",
        "arg": 4,
        "term": 117
    },
    {
        "opcode": "load",
        "arg": 4,
        "term": 118
    },
    {
        "opcode": "addc",
        "arg": -2,
        "term": 119
    },
    {
        "opcode": "jne",
        "arg": 123,
        "term": 120
    },
    {
        "opcode": "load",
        "arg": 5,
        "term": 121
    },
    {
        "opcode": "je",
        "arg": 125,
        "term": 122
    },
    {
        "opcode": "jmp",
        "arg": 126,
        "term": 123
    },
    {
        "opcode": "jmp",
        "arg": 125,
        "term": 124
    },
    {
        "opcode": "jmp",
        "arg": 12,
        "term": 125
    },
    {
        "opcode": "load",
        "arg": 2,
        "term": 126
    },
    {
        "opcode": "store",
        "arg": 4,
        "term": 127
    },
    {
        "opcode": "load",
        "arg": 3,
        "term": 128
    },
    {
        "opcode": "store",
        "arg": 5,
        "term": 129
    },
    {
        "opcode": "load",
        "arg": 5,
        "term": 130
    },
    {
        "opcode": "store",
        "arg": 6,
        "term": 131
    },
    {
        "opcode": "load",
        "arg": 4,
        "term": 132
    },
    {
        "opcode": "je",
        "arg": 147,
        "term": 133
    },
    {
        "opcode": "addc",
        "arg": -1,
        "term": 134
    },
    {
        "opcode": "je",
        "arg": 157,
        "term": 135
    },
    {
        "opcode": "addc",
        "arg": -1,
        "term": 136
    },
    {
        "opcode": "je",
        "arg": 145,
        "term": 137
    },
    {
        "opcode": "mem",
        "arg": 6,
        "term": 138
    },
    {
        "opcode": "je",
        "arg": 159,
        "term": 139
    },
    {
        "opcode": "print",
        "arg": "",
        "term": 140
    },
    {
        "opcode": "load",
        "arg": 6,
        "term": 141
    },
    {
        "opcode": "addc",
        "arg": 1,
        "term": 142
    },
    {
        "opcode": "store",
        "arg": 6,
        "term": 143
    },
    {
        "opcode": "jmp",
        "arg": 138,
        "term": 144
    },
    {
        "opcode": "load",
        "arg": 6,
        "term": 145
    },
    {
        "opcode": "jne",
        "arg": 154,
        "term": 146
    },
    {
        "opcode": "loadc",
        "arg": 78,
        "term": 147
    },
    {
        "opcode": "print",
        "arg": "",
        "term": 148
    },
    {
        "opcode": "loadc",
        "arg": 73,
        "term": 149
    },
    {
        "opcode": "print",
        "arg": "",
        "term": 150
    },
    {
        "opcode": "loadc",
        "arg": 76,
        "term": 151
    },
    {
        "opcode": "print",
        "arg": "",
        "term": 152
    },
    {
        "opcode": "jmp",
        "arg": 159,
        "term": 153
    },
    {
        "opcode": "loadc",
        "arg": 116,
        "term": 154
    },
    {
        "opcode": "print",
        "arg": "",
        "term": 155
    },
    {
        "opcode": "jmp",
        "arg": 159,
        "term": 156
    },
    {
        "opcode": "load",
        "arg": 6,
        "term": 157
    },
    {
        "opcode": "print_int",
        "arg": "",
        "term": 158
    },
    {
        "opcode": "loadc",
        "arg": 10,
        "term": 159
    },
    {
        "opcode": "print",
        "arg": "",
        "term": 160
    },
    {
        "opcode": "halt",
        "arg": "",
        "term": 161
    }
]