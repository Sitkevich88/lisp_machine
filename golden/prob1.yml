source: |-
  ( setq result 0 )

  (loop for i from 0 to 999 do
      ( if
           ( = ( mod i 3 ) 0 )
           ( setq result ( + result i ) )
           ( if
                ( = ( mod i 5 ) 0 )
                ( setq result ( + result i ) )
            )
      )
  )

  ( print result )
input: |

code: |

output: |

log: |
